controllers/target_status_controller_test.go

```diff
@@ -0,0 +1,560 @@
+							Authentication: map[string]v1beta2.AuthenticationSpec{
+								"anonymous": {
+									AuthenticationSpec: authorinoapi.AuthenticationSpec{
+										AuthenticationMethodSpec: authorinoapi.AuthenticationMethodSpec{
+											AnonymousAccess: &authorinoapi.AnonymousAccessSpec{},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			for _, mutateFn := range mutateFns {
+				mutateFn(policy)
+			}
+			return policy
+		}
+
+		// policyAcceptedAndTargetsAffected returns an assertion function that checks if an AuthPolicy is accepted
+		// and the statuses of its target object and other optional route objects have been all updated as affected by the policy
+		policyAcceptedAndTargetsAffected := func(policy *v1beta2.AuthPolicy, routeNames ...string) func() bool {
+			return func() bool {
+				if !isAuthPolicyAccepted(policy)() {
+					return false
+				}
+				return targetsAffected(client.ObjectKeyFromObject(policy), policyAffectedCondition, policy.Spec.TargetRef, routeNames...)
+			}
+		}
+
+		It("adds PolicyAffected status condition to the targeted route", func() {
+			policy := policyFactory()
+			Expect(k8sClient.Create(context.Background(), policy)).To(Succeed())
+			Eventually(policyAcceptedAndTargetsAffected(policy), 30*time.Second, 5*time.Second).Should(BeTrue())
+		})
+
+		It("removes PolicyAffected status condition from the targeted route when the policy is deleted", func() {
+			policy := policyFactory()
+			Expect(k8sClient.Create(context.Background(), policy)).To(Succeed())
+			Eventually(policyAcceptedAndTargetsAffected(policy), 30*time.Second, 5*time.Second).Should(BeTrue())
+
+			Expect(k8sClient.Delete(context.Background(), policy)).To(Succeed())
+
+			Eventually(func() bool { // route is not affected by the policy
+				route := &gatewayapiv1.HTTPRoute{}
+				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: testHTTPRouteName, Namespace: testNamespace}, route)
+				if err != nil {
+					return false
+				}
+				routeParentStatus, found := utils.Find(route.Status.RouteStatus.Parents, findRouteParentStatusFunc(route, client.ObjectKey{Name: testGatewayName, Namespace: testNamespace}, kuadrant.ControllerName))
+				return !found || meta.IsStatusConditionFalse(routeParentStatus.Conditions, policyAffectedCondition)
+			}, 30*time.Second, 5*time.Second).Should(BeTrue())
+		})
+
+		It("adds PolicyAffected status condition to the targeted gateway and routes", func() {
+			policy := policyFactory(func(policy *v1beta2.AuthPolicy) {
+				policy.Name = "gateway-auth"
+				policy.Spec.TargetRef = gatewayapiv1alpha2.PolicyTargetReference{
+					Group:     gatewayapiv1.GroupName,
+					Kind:      "Gateway",
+					Name:      testGatewayName,
+					Namespace: ptr.To(gatewayapiv1.Namespace(testNamespace)),
+				}
+			})
+			Expect(k8sClient.Create(context.Background(), policy)).To(Succeed())
+			Eventually(policyAcceptedAndTargetsAffected(policy), 30*time.Second, 5*time.Second).Should(BeTrue())
```

Comment:
Should the toystore route name also be passed here and possibly a few other places to check that GW policies put the affected condition to underlying untargeted routes ? 🤔 
```suggestion
			Eventually(policyAcceptedAndTargetsAffected(policy, testHTTPRouteName), 30*time.Second, 5*time.Second).Should(BeTrue())
```

Reply:
Nice catch! Fixed.